x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java

```diff
@@ -606,6 +621,74 @@ private void setMetadataFileWatcher(String jwkSetPath) throws IOException {
         watcherService.add(watcher, ResourceWatcherService.Frequency.MEDIUM);
     }
 
+    /**
+     * Merges the JsonObject with the claims of the ID Token with the JsonObject with the claims of the UserInfo response. This is
+     * necessary as some OPs return slightly different values for some claims (i.e. Google for the profile picture) and
+     * {@link JSONObject#merge(Object)} would throw a runtime exception. The merging is performed based on the following rules:
+     * <ul>
+     * <li>If the values for a given claim are strings, the value from the ID Token is retained</li>
```

Comment:
Can a token not also have numbers and booleans?
```suggestion
     * <li>If the values for a given claim are primitives (of the the same type), the value from the ID Token is retained</li>
```